Volume/Views/MainView/MagazinesList.swift

```diff
@@ -10,23 +10,50 @@ import Combine
 import SwiftUI
 
 struct MagazinesList: View {
-    @State private var cancellableQuery: AnyCancellable?
-    @State private var state: MainView.TabState<Results> = .loading
     @EnvironmentObject private var networkState: NetworkState
+    @State private var onOpenMagazineUrl: String?
+    @State private var openedUrl = false
+    @State private var sectionQueries: SectionQueries = (nil, nil)
+    @State private var sectionStates: SectionStates = (.loading, .loading)
+
 
     private func fetchContent(_ done: @escaping () -> Void = { }) {
-        // Simulate network request delay with Publication network request
-        // TODO: Replace with Magazines request
-        cancellableQuery = Network.shared.publisher(for: GetAllPublicationsQuery())
-            .map { data in data.publications.compactMap { $0 } }
+        guard sectionStates.featuredMagazines.isLoading || sectionStates.otherMagazines.isLoading else { return }
+        
+        fetchFeaturedMagazines(done)
+        fetchOtherMagazines()
+    }
+    
+    private func fetchFeaturedMagazines(_ done: @escaping () -> Void = { }) {
+        sectionQueries.featuredMagazines = Network.shared.publisher(for: GetFeaturedMagazinesQuery(limit: 7))
+            .map {
+                // not sure if force-unwrapping here is safe
+                $0.magazines!.map(\.fragments.magazineFields)
```

Comment:
Delete this comment too. I would try to not force unwrap if it you can. @dishanest thoughts?

Reply:
Chain the optionals